@jirikuncar any idea? :)

```
Warning, treated as error:

/home/travis/build/inveniosoftware/invenio-jsonschemas/docs/api.rst:28: WARNING: autodoc: failed to import module u'invenio_jsonschemas.ext'; the following exception was raised:

Traceback (most recent call last):

  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 519, in import_object

    __import__(self.modname)

ImportError: No module named invenio_jsonschemas.ext
```

--- a/invenio_jsonschemas/config.py
+++ b/invenio_jsonschemas/config.py
@@ -29,3 +29,6 @@ JSONSCHEMAS_HOST = 'localhost'
 
 JSONSCHEMAS_ENDPOINT = '/schemas'
 """Default schema endpoint."""
+
+JSONSCHEMAS_URL_SCHEME = 'http'
+"""Default url scheme for schemas."""

--- a/invenio_jsonschemas/ext.py
+++ b/invenio_jsonschemas/ext.py
@@ -131,7 +131,7 @@ class InvenioJSONSchemasState(object):
     def list_schemas(self):
         """List all JSON-schema names.
 
-        :Return: list of schema names.
+        :returns: list of schema names.
         :rtype: list
         """
         return self.schemas.keys()
@@ -172,8 +172,10 @@ class InvenioJSONSchemas(object):
     Register blueprint serving registered schemas and can be used as an api
     to register those schemas.
 
-    Note: JSON schemas are served as static files. Thus their "id" and "$ref"
-    fields might not match the Flask application's host and port.
+    .. note::
+
+        JSON schemas are served as static files. Thus their "id" and "$ref"
+        fields might not match the Flask application's host and port.
     """
 
     def __init__(self, app=None, **kwargs):

--- a/invenio_jsonschemas/ext.py
+++ b/invenio_jsonschemas/ext.py
@@ -159,7 +159,10 @@ class InvenioJSONSchemasState(object):
         """
         if path not in self.schemas:
             return None
-        return self.url_map.bind(self.app.config['JSONSCHEMAS_HOST']).build(
+        return self.url_map.bind(
+            self.app.config['JSONSCHEMAS_HOST'],
+            url_scheme=self.app.config['JSONSCHEMAS_URL_SCHEME']
+        ).build(
             'schema', values={'path': path}, force_external=True)
 
 

--- a/tests/test_examples_app.py
+++ b/tests/test_examples_app.py
@@ -59,6 +59,8 @@ def example_app():
     subprocess.call(cmd, shell=True)
     # return to the original directory
     os.chdir(current_dir)
+    cmd = 'pip install -e .[all]'
+    subprocess.call(cmd, shell=True)
 
 
 def test_example_app(example_app):

--- a/tests/test_invenio_jsonschemas.py
+++ b/tests/test_invenio_jsonschemas.py
@@ -39,6 +39,7 @@ from jsonschema import validate
 from jsonschema.exceptions import ValidationError
 
 from invenio_jsonschemas import InvenioJSONSchemas
+from invenio_jsonschemas.config import JSONSCHEMAS_URL_SCHEME
 from invenio_jsonschemas.errors import JSONSchemaDuplicate, JSONSchemaNotFound
 
 
@@ -111,6 +112,7 @@ def test_api(app, dir_factory):
 
 class mock_open(object):
     """Mock the builtin 'open' and count the file requests."""
+
     counter = 0
 
     def __init__(self, path):
@@ -274,10 +276,18 @@ def test_jsonresolver():
         assert exc_info.value.schema == {'type': 'integer'}
 
 
-def test_url_mapping(app, dir_factory):
+@pytest.mark.parametrize('url_scheme', [
+    None, 'http', 'https'
+])
+def test_url_mapping(app, dir_factory, url_scheme):
     """Test register schema."""
     app.config['SERVER_NAME'] = 'example.org'
     app.config['JSONSCHEMAS_HOST'] = 'inveniosoftware.org'
+    if url_scheme is not None:
+        app.config['JSONSCHEMAS_URL_SCHEME'] = url_scheme
+    else:
+        # test with default url scheme configuration
+        url_scheme = JSONSCHEMAS_URL_SCHEME
 
     ext = InvenioJSONSchemas(app, entry_point_group=None)
     schema_files = build_schemas(1)
@@ -286,13 +296,17 @@ def test_url_mapping(app, dir_factory):
         ext.register_schemas_dir(directory)
         with app.app_context():
             assert 'sub1/subschema_1.json' == ext.url_to_path(
-                'http://inveniosoftware.org/schemas/sub1/subschema_1.json')
+                '{0}://inveniosoftware.org/schemas/sub1/subschema_1.json'
+                .format(url_scheme))
             assert ext.url_to_path(
-                'http://inveniosoftware.org/schemas/invalid.json') is None
+                '{0}://inveniosoftware.org/schemas/invalid.json'
+                .format(url_scheme)) is None
             assert ext.url_to_path(
-                'http://example.org/schemas/sub1/subschema_1.json') is None
+                '{0}://example.org/schemas/sub1/subschema_1.json'
+                .format(url_scheme)) is None
 
             assert (
-                'http://inveniosoftware.org/schemas/sub1/subschema_1.json'
+                '{0}://inveniosoftware.org/schemas/sub1/subschema_1.json'
+                .format(url_scheme)
             ) == ext.path_to_url('sub1/subschema_1.json')
             assert ext.path_to_url('invalid.json') is None

